### PR TITLE
*: add schema related errors FAQ and description of `tidb_max_delta_schema_count`

### PR DESCRIPTION
**Pull request by zimulala**
### What is changed, added or deleted? 
Add schema related errors FAQ and description of `tidb_max_delta_schema_count`.

### Which version does your change affect? 
v2.1, v3.0 and dev

### CODE REVIEW

File: dev/reference/configuration/tidb-server/tidb-specific-variables.md [hunk 1]
```diff
@@ -421,6 +421,14 @@ set @@global.tidb_distsql_scan_concurrency = 10;
 
 这个变量用来控制 DDL 操作失败重试的次数。失败重试次数超过该参数的值后，会取消出错的 DDL 操作。
 
+### tidb_max_delta_schema_count
+
+作用域：GLOBAL
+
+默认值：1024
```
Comment (dcalvin) [inline dev/reference/configuration/tidb-server/tidb-specific-variables.md]: Inconsistent default value as shown below.

> 个数限制，默认值为 100

File: dev/faq/tidb.md [hunk 2]
```diff
@@ -566,13 +566,20 @@ TiDB 在执行 SQL 语句时，会使用当时的 `schema` 来处理该 SQL 语
...
 > **注意：**
 >
 > `create table` 操作会有 1 个 `schema` 版本变更，与每个 DDL 操作对应变更的 `schema state` 个数一致。例如，`add column` 操作会有 4 个版本信息。
 
+#### 3.3.14 触发 Information schema is out of date 错误的原因？
+
+当执行 DML 时，TiDB 超过一个 DDL lease 时间（默认 45s）没能加载到最新的 schema 就可能会报 `Information schema is out of date` 的错误。遇到此错的可能原因如下：
+
+- 执行此 DML 的 TiDB 被 kill 后准备退出，且此 DML 对应的事务执行时间超过一个 DDL lease，在事务提交时会报这个错误。
+- TiDB 在执行此 DML 时，有一段时间内连不上 PD 或者 TiKV。导致 TiDB 超过一个 DDL lease 时间没有 load schema，或者导致 TiDB 断开与 PD 的带 keep alive 的连接。
```
Comment (dcalvin) [inline dev/faq/tidb.md]:
```suggestion
- TiDB 在执行此 DML 时，有一段时间内连不上 PD 或者 TiKV，导致 TiDB 超过一个 DDL lease 时间没有 load schema，或者导致 TiDB 断开与 PD 之间带 keep alive 设置的连接。
```

**Comment by zimulala** [comment]
PTAL @dcalvin @crazycs520

**Comment by lilin90** [comment]
@crazycs520 PTAL

File: dev/faq/tidb.md [hunk 3]
```diff
@@ -566,13 +566,20 @@ TiDB 在执行 SQL 语句时，会使用当时的 `schema` 来处理该 SQL 语
 
 - 执行的 DML 语句中涉及的表和集群中正在执行的 DDL 的表有相同的，那么这个 DML 语句就会报此错。
 - 与表无关的报错原因：
-    - 这个 DML 执行时间很久，而这段时间内执行了很多 DDL 语句（新版本 `lock table` 也可），导致中间 `schema` 版本变更超过 1024
+    - 这个 DML 执行时间很久，而这段时间内执行了很多 DDL 语句（新版本 `lock table` 也可），导致中间 `schema` 版本变更超过 1024 (此为默认值，可以通过 `tidb_max_delta_schema_count` 变量修改)。
```
Comment (dcalvin) [inline dev/faq/tidb.md]: > 新版本 `lock table` 也可

What does this mean?

Comment (zimulala) [reply dev/faq/tidb.md]: lock table 语句也可以

**Pull request merged**